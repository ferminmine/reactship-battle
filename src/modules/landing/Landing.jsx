import React from 'react';
import { connect } from 'react-redux';
import { compose, bindActionCreators } from 'redux';
import { setGameState } from '../game/GameActions';
import { setPlayerName } from '../player/PlayerActions';
import GAME_STATES from '../game/GameStates';
import { nextShipToPlaceSelector } from '../ship/ShipSelectors';
import ShipPlacement from '../ship/ShipPlacement/ShipPlacement';
import withStyles from 'react-jss';
import ShipToPlace from '../ship/ShipPlacement/ShipToPlace';
import PropTypes from 'prop-types';

const styles = {
  shipPlacementContainer: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: '2em'
  },
  optionsContainer: {
    marginLeft: '3%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  nameInput: {
    border: '1px solid #b4bdbc',
    borderRadius: 0,
    fontSize: '1.1em',
    padding: '10px',
    textAlign: 'center'
  },
  startButton: {
    marginTop: '2em',
    width: '70%',
    background: 'none',
    border: '3px solid #899695',
    color: '#899695',
    fontWeight: '800',
    padding: '10px',
    borderRadius: '6px',
    fontSize: '1.1em',
    cursor: 'pointer'
  },
  disabledButton: {
    backgroundColor: '#b9b9b9',
    pointerEvents: 'none'
  }
};

class Landing extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    game: PropTypes.object.isRequired,
    setGameState: PropTypes.func.isRequired,
    setPlayerName: PropTypes.func.isRequired
  };

  startGame = () => {
    if (this.props.game.state === GAME_STATES.PLAYER_READY) {
      this.props.history.push('/play');
    }
  };

  onChangeName = event => {
    this.props.setPlayerName(event.target.value);
  };

  componentDidUpdate = () => {
    if (
      this.props.player !== '' &&
      this.props.nextShipToPlace === undefined &&
      this.props.game.state !== GAME_STATES.PLAYER_READY
    ) {
      this.props.setGameState(GAME_STATES.PLAYER_READY);
    }
  };

  render = () => {
    const { classes, game } = this.props;
    const buttonState = game.state === GAME_STATES.PLAYER_STARTING ? 'disabledButton' : ''
    return (
      <div className={classes.shipPlacementContainer}>
        <ShipPlacement />
        <div className={classes.optionsContainer}>
          <ShipToPlace />
          <input
            type="text"
            className={classes.nameInput}
            onChange={this.onChangeName}
            placeholder="Enter Your Name"
          />
          <button className={`${classes.startButton} ${classes[buttonState]}`} onClick={this.startGame}>
            {' '}
            START GAME{' '}
          </button>
        </div>
      </div>
    );
  };
}

const mapStateToProps = state => ({
  game: state.game,
  player: state.player,
  nextShipToPlace: nextShipToPlaceSelector(state)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setGameState,
      setPlayerName
    },
    dispatch
  );

export default compose(
  withStyles(styles),
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(Landing);
