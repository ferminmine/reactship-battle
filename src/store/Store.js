import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import playerReducer from '../modules/player/PlayerReducer';
import shipReducer from '../modules/ship/ShipReducer';
import gameReducer from '../modules/game/GameReducer';
import shootsReducer from '../modules/shoots/ShootsReducer';

export const history = createBrowserHistory();

const allReducers = combineReducers({
  player: playerReducer,
  game: gameReducer,
  ships: shipReducer,
  shoots: shootsReducer,
  router: connectRouter(history)
});

export const ROOT_REDUCER_ACTION_TYPES = {
  RESET_APP: 'RESET_APP'
};
export const resetApp = () => ({ type: ROOT_REDUCER_ACTION_TYPES.RESET_APP });

const rootReducer = (state, action) => {
  if (action.type === 'RESET_APP') {
    state = undefined;
  }

  return allReducers(state, action);
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default createStore(
  rootReducer,
  composeEnhancer(applyMiddleware(routerMiddleware(history)))
);
