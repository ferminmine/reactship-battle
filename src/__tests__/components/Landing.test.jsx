import React from 'react';
import { mount } from 'enzyme';
import Landing from '../../modules/landing/Landing';
import { Provider } from 'react-redux';
import store from '../../store/Store';
import renderer from 'react-test-renderer';

describe('Landing Component Test', () => {
  it('renders without crashing', () => {
    mount(
      <Provider store={store}>
        <Landing />
      </Provider>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Landing />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
