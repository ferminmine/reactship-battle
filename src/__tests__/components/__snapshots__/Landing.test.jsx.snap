// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component Test matches the snapshot 1`] = `
Array [
  <div
    className="ParagraphWithTitle-instructionsContainer-0-1-6"
  >
    <div
      className="ParagraphWithTitle-instructionsParagraph-0-1-7"
    >
      <div
        className="ParagraphWithTitle-paragraphTitle-0-1-8"
      >
         
        What Is Reactship Battle?
         
      </div>
      It is the good and old Battleship game that you probably know, but implemented for the browser
so you can play anytime you want against our well polished AI who will be delighted to play
against you. Be warned though: it will not give you any mercy.
    </div>
  </div>,
  <div
    className="ParagraphWithTitle-instructionsContainer-0-1-6"
  >
    <div
      className="ParagraphWithTitle-instructionsParagraph-0-1-7"
    >
      <div
        className="ParagraphWithTitle-paragraphTitle-0-1-8"
      >
         
         
      </div>
      If you have not played Battleship before, then you can read the rules
       
      <a
        href="https://en.wikipedia.org/wiki/Battleship_(game)"
        rel="noopener noreferrer"
        target="_blank"
      >
        here
      </a>
      .
    </div>
  </div>,
  <div
    className="Connect(Landing)-shipPlacementContainer-0-1-1"
  >
    <div
      className="Connect(ShipPlacement)-buttonContainer-0-1-9"
    >
      <div
        className="Connect(ShipPlacement)-rotateButton-0-1-10"
        onClick={[Function]}
      >
         ROTATE SHIP 
      </div>
    </div>
    <div
      className="Connect(ShipPlacement)-boardOutsideContainer-0-1-11"
    >
      <div
        className="Connect(Board)-boardContainer-0-1-12"
      >
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <div
          className="Connect(Board)-land-0-1-13 undefined undefined undefined undefined undefined undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      </div>
    </div>
    <div
      className="Connect(Landing)-optionsContainer-0-1-2"
    >
      <div
        className="Connect(ShipToPlace)-shipToPlaceContainer-0-1-20"
      >
        <h1
          className="Connect(ShipToPlace)-nextShip-0-1-21"
        >
          Next Ship to Place
        </h1>
        <img
          alt=""
          className="Connect(ShipToPlace)-shipImage-0-1-23"
          src="/imgs/cruiser.png"
        />
        <h2
          className="Connect(ShipToPlace)-shipName-0-1-22"
        >
           
          Cruiser
           
        </h2>
      </div>
      <input
        className="Connect(Landing)-nameInput-0-1-3"
        onChange={[Function]}
        placeholder="Enter Your Name"
        type="text"
      />
      <button
        className="Connect(Landing)-startButton-0-1-4 Connect(Landing)-disabledButton-0-1-5"
        onClick={[Function]}
      >
         
        START GAME
         
      </button>
    </div>
  </div>,
]
`;
